wdl/PickleVcfProperties.wdl

```diff
@@ -0,0 +1,121 @@
+version development
+
+import "GetTruthOverlap.wdl" as GetTruthOverlap
+
+workflow PickleVcfProperties {
+    input {
+        File vcf
+        File? vcf_index
+        Array[String] wanted_properties
+        String sv_base_docker
+        String sv_utils_docker
+    }
+
+    call GetTruthOverlap.GetVcfSize as GetVcfSize {
+        input:
+            vcf=vcf,
+            vcf_index=vcf_index,
+            sv_base_docker=sv_base_docker
+    }
+
+    call GetNeededMemGB {
+        input:
+            wanted_properties=wanted_properties,
+            num_entries=GetVcfSize.num_entries
+    }
+
+    call ReadAndPickleProperties {
+        input:
+          vcf=vcf,
+          wanted_properties=wanted_properties,
+          mem_gb=GetNeededMemGB.pickle_task_mem_gb,
+          sv_utils_docker=sv_utils_docker
+    }
+
+    output {
+        Int num_records = GetVcfSize.num_records
+        Int num_samples = GetVcfSize.num_samples
+        Int num_entries = GetVcfSize.num_entries
+        File pickled_properties = ReadAndPickleProperties.pickled_properties
+    }
+}
+
+task GetNeededMemGB {
```

Comment:
It shouldn't have to be a task, but due to the limitations of WDL, there's no way (that I can find) to check if a key is present amongst the wanted_properties, nor to use a default value if it is not in the cost map. It seemed like an acceptable solution because it's a super quick task in a cheap VM. Should cost much wall clock or $$. And there are very big differences in the VM sizing (due to the number of entries, and a big jump in required memory if there are any missing entries). So I'd much rather have this task than skip it and just size all the pickle tasks the same.

Reply:
It shouldn't have to be a task, but due to the limitations of WDL, there's no way (that I can find) to check if a key is present amongst the wanted_properties, nor to use a default value if it is not in the cost map. It seemed like an acceptable solution because it's a super quick task in a cheap VM. Should cost much wall clock or $$. And there are very big differences in the VM sizing (due to the number of entries, and a big jump in required memory if there are any missing entries). So I'd much rather have this task than skip it and just size all the pickle tasks the same.